support-frontend/app/controllers/Application.scala

```diff
@@ -49,6 +49,13 @@ class Application(
     Ok(views.html.contributionsRedirect())
   }
 
+  //Encode the querystring parameter keys, as Play only encodes the values
+  def redirectWithEncodedQueryString(url: String, queryString: Map[String, Seq[String]] = Map.empty, status: Int = SEE_OTHER): Result = Redirect(
+    url = url,
+    queryString = queryString.map { case (k,v) => java.net.URLEncoder.encode(k, "utf-8") -> v },
```

Comment:
have you tested that an actually encoded input e.g. %7f in the incoming url, is decoded in the "key" and then gets reencoded suitably on the way out.  And that it doesn't get double encoded.
This seems to document the "incoming" side, which suggests the decoding is done as per RFC3986:
https://doc.akka.io/docs/akka-http/current/common/uri-model.html
From reading those, I'd guess yes, but I haven't got that far yet.

Reply:
Yes, the screenshot does kind of show this, but we can confirm with:
`curl "https://support.thegulocal.com/contribute?INTCMP=side_menu_support_contribute&acquisitionData=%7B&quo.%E2%80%99" -v`
which returns:
```
< HTTP/1.1 302 Found
...
< location: /uk/contribute?INTCMP=side_menu_support_contribute&acquisitionData=%7B&quo.%E2%80%99=
...
```
The additional `=` at the end of the `location` header is because it's interpreted as a querystring parameter with an empty value

Reply:
I see, so the akka/play query string parser assumes the query string is encoded using https://url.spec.whatwg.org/#urlencoded-serializing
This means it interprets the & as the start of a new key and the lack of an `=` as an empty value.  I'm not sure there's much we can do about it, assuming we're going to build the query string using the same mechanism.  But hopefully we wouldn't rely on that not happening.